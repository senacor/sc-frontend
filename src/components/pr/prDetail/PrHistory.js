import React, { Component } from 'react';

import TableHead from '@material-ui/core/TableHead/TableHead';
import TableRow from '@material-ui/core/TableRow/TableRow';
import TableCell from '@material-ui/core/TableCell/TableCell';
import TableBody from '@material-ui/core/TableBody/TableBody';
import Table from '@material-ui/core/Table/Table';

class PrHistory extends Component {
  constructor(props) {
    super(props);

    this.state = {
      prHistory: [
        { id: 5, appointment: '2018-01-01', occassion: 'ON_DEMAND' },
        {
          id: 6,
          isExcel: true,
          filename: '20170101_Michaela_Mitarbeiterin.xls'
        },
        {
          id: 7,
          isExcel: true,
          filename: '20160101_Michaela_Mitarbeiterin.xls'
        }
      ]
    };
  }
  render() {
    return (
      <div>
        <Table padding={'none'}>
          <TableHead>
            <TableRow>
              <TableCell padding={'none'}>Datum</TableCell>
              <TableCell padding={'none'}>Link</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {this.state.prHistory.map(row => {
              return (
                <TableRow key={row.id}>
                  <TableCell padding={'none'}>
                    {row.appointment ? row.appointment : 'Excel'}
                  </TableCell>
                  <TableCell padding={'none'}>
                    {row.occassion
                      ? row.occassion
                      : `Download Excel ${row.filename}`}
                  </TableCell>
                </TableRow>
              );
            })}
          </TableBody>
        </Table>
      </div>
    );
  }
}

export default PrHistory;
