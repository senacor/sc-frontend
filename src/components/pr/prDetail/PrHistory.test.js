import React from 'react';
import PrHistory from './PrHistory';
import { shallow } from 'enzyme';

describe('PrHistory Component', () => {
  it('should match snapshot', () => {
    const component = shallow(<PrHistory />);
    expect(component).toMatchSnapshot();
  });
});
