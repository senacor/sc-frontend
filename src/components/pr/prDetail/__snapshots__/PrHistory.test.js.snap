// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrHistory Component should match snapshot 1`] = `
<div>
  <WithStyles(Table)
    padding="none"
  >
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          padding="none"
        >
          Datum
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          padding="none"
        >
          Link
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)
        key="5"
      >
        <WithStyles(TableCell)
          padding="none"
        >
          2018-01-01
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          padding="none"
        >
          ON_DEMAND
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        key="6"
      >
        <WithStyles(TableCell)
          padding="none"
        >
          Excel
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          padding="none"
        >
          Download Excel 20170101_Michaela_Mitarbeiterin.xls
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        key="7"
      >
        <WithStyles(TableCell)
          padding="none"
        >
          Excel
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          padding="none"
        >
          Download Excel 20160101_Michaela_Mitarbeiterin.xls
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</div>
`;
